native/src/main/java/io/ballerina/stdlib/java.jms/producer/Actions.java

```diff
@@ -99,21 +101,19 @@ public static Object send(Environment env, BObject producer, Message message) {
      * Sends a message to a destination for an unidentified message producer using the {@code MessageProducer}'s
      * default delivery mode, priority, and time to live.
      *
-     * @param env Ballerina runtime environment
      * @param producer    Ballerina producer object
-     * @param session     Ballerina session object
      * @param destination Relevant JMS destination
-     * @param message     The JMS message
+     * @param bMessage  The Ballerina JMS message representation
      * @return A Ballerina `jms:Error` if the JMS MessageProducer fails to send the message due to some error
      */
-    public static Object sendTo(Environment env, BObject producer, BObject session, BMap<BString, Object> destination,
-                                Message message) {
+    public static Object sendTo(BObject producer, BMap<BString, Object> destination, BMap<BString, Object> bMessage) {
         MessageProducer nativeProducer = (MessageProducer) producer.getNativeData(NATIVE_PRODUCER);
-        Session nativeSession = (Session) session.getNativeData(NATIVE_SESSION);
+        Session nativeSession = (Session) producer.getNativeData(NATIVE_SESSION);
         CompletableFuture<Object> balFuture = new CompletableFuture<>();
```

Comment:
Ditto. I think this is an issue when we migrated from Java17 to Java 21.

Reply:
NVM. Found the usage. But still I think we can improve this logic (without using a future), but can do that later.